app/user.go

```diff
@@ -1025,17 +1025,28 @@ func (a *App) UpdateUserAuth(userID string, userAuth *model.UserAuth) (*model.Us
 
 func (a *App) sendUpdatedUserEvent(user model.User) {
 	adminCopyOfUser := user.DeepCopy()
+	unsanitizedCopyOfUser := user.DeepCopy()
+
 	a.SanitizeProfile(adminCopyOfUser, true)
 	adminMessage := model.NewWebSocketEvent(model.WebsocketEventUserUpdated, "", "", "", nil)
 	adminMessage.Add("user", adminCopyOfUser)
 	adminMessage.GetBroadcast().ContainsSensitiveData = true
 	a.Publish(adminMessage)
 
+	// exclude event creator user from broadcast
+	omitUsers := make(map[string]bool, 1)
+	omitUsers[user.Id] = true
 	a.SanitizeProfile(&user, false)
-	message := model.NewWebSocketEvent(model.WebsocketEventUserUpdated, "", "", "", nil)
+	message := model.NewWebSocketEvent(model.WebsocketEventUserUpdated, "", "", "", omitUsers)
 	message.Add("user", &user)
 	message.GetBroadcast().ContainsSanitizedData = true
 	a.Publish(message)
+
+	// send unsanitized user to event creator
+	sourceUserMessage := model.NewWebSocketEvent(model.WebsocketEventUserUpdated, "", "", unsanitizedCopyOfUser.Id, nil)
```

Comment:
correct me if I'm wrong, but I'm seeing 3 events being published here, shouldn't we ensure that the user only receives the one event.

What I see here is
`adminMessage` being broadcasted to everyone (this seems wrong, shouldn't this omit the source creator as well?)
`message` being broadcasted excluding the source creator (this looks ok to me)
`sourceUserMessage` being broadcasted only to the source creator (this looks ok to me)

Reply:
I think https://github.com/mattermost/mattermost-server/blob/master/app/web_conn.go#L721 forwards to all admins, so I get two events (one `sourceUserMessage ` and one `adminMessage`) when admin is the data source, thanks for pointing this out. I'll omit `sourceUser` from adminMessage.

Reply:
Please let me know if the latest commit fixes it?

Reply:
Ah my apologies on that one. You are correct @enahum. Great catch!